### PR TITLE
Python toolbox script to maintain the Metrobike dropbox account when it's full

### LINKED ISSUE #17077: Backup Metrobike Dropbox data
Our contact at Metrobike, Hunter, is saying the dropbox folder we are using for metrobike is full and we need to delete things before we can add new data. It would be good to backup this data in a S3 bucket before we delete anything that way we have a copy of the historical data.

We don't have an account for dropbox as far as I know, so we use the metrobike dropbox API key (in 1password).

### PR DESCRIPTION
**Pull request by frankhereford**
# Intent

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/17077.

It introduces a Python script clean_out_dropbox.py that provides several utilities for managing files in a Dropbox account. The script uses the Dropbox and AWS S3 SDKs to list, mirror, and delete files.

# Additional Caveat

* The API key we have is a RO key, so the delete functionality is unavailable at this time.

# Usage

* Copy the `env_template` into `env` and complete it. The dropbox token is in 1PW and the AWS credentials are your own.

```bash
docker compose build
docker compose run clean_up_dropbox 
```
* You'll be dropped at a bash prompt and you can run the script like this:

```bash
./clean_out_dropbox.py -h
```

# Testing

* Use the script.
  * You can use any combination of the `-i`, `-m`, and `-s` flags in a non-destructive way with no fear, but the `-d` flag will empty out the dropbox folder ⚠️. We'd have to write a new script to put stuff back in there, so don't use `-d` unless you intend for there to be deletions. ⚠️

## Command line options:

`-i` or `--inspect`: Lists all files in the Dropbox account and their sizes.
`-m` or `--mirror`: Mirrors the contents of the Dropbox account to a local directory named dropbox_mirror.
`-s` or `--mirror-to-s3`: Mirrors the contents of the Dropbox account to an S3 bucket. The bucket name is hardcoded as metrobike-historic-data and the files are stored in a new folder with a timestamp.
`-d` or `--delete`: ⚠️ Deletes all files in the Dropbox account after a confirmation step. The user is asked to type a randomly generated 4-digit number to confirm the deletion.

### CODE REVIEW

**Comment by frankhereford** [comment]
@Charlie-Henry I laughed a little about this -- I wrote the script and the last part that I did was to add the destructive deleting capability which is the part that does the real core of this issue's request, and it turns out that the token we have is a read-only token. So, the delete functionality of the script will fail and throw an exception until we can feed it a RW token. 

But it does work pretty good to inspect and mirror the data out of dropbox, either into a local folder or a S3 bucket, so that's pretty good. Let's sync on Monday or whenever you have a good time. Thanks!

File: toolbox/clean_out_dropbox/.dockerignore
Comment (frankhereford) [inline toolbox/clean_out_dropbox/.dockerignore:1]: This looks funny in the diff, but this is a symlink to `.gitignore`. Whatever git ignores, docker ignores.

**Review by Charlie-Henry — APPROVED** [review]
Ah I should've figured about the read only API key. I'll reach out to Hunter over at cap metro and let him know we've backed things up on our side. I don't think we need the ability to delete things but we'll see what he says.

Looks good to me! I didn't run things but I do see the backed up files in S3!

**Review by chiaberry — APPROVED** [review]
looks good, do you think we should include a note to our future selves somewhere that we don't have a key with delete access?

**Comment by frankhereford** [comment]
@chiaberry wrote:

> looks good, do you think we should include a note to our future selves somewhere that we don't have a key with delete access?

The program raises an exception in a non-destructive way if it's given a RO key and asked to delete, so I think we're probably good, but I will add a note in the readme to note that. If we have any documentation about our metrobike relationship, that may be a good place too. I'll defer to Charlie on what he thinks we should do outside of this PR though.

**Pull request merged**